### PR TITLE
feat (networking): [streamline] remove hub and move la and bastion

### PR DESCRIPTION
**Pull request by ferantivero**
related: #83068

## WHY

we wanted to have a simplified version of the initial iaas baseline that could help enhancing our focus over compute rather than networking. For that we wanted to remove initially hub and spoke network topology. This way, this RI will help to minimize the amount of resources being deployed and lessons to be initially ingested/learned.  

## WHAT Changed?

- remove the hub networking assets
- change docs to deploy the spoke onky
- change the spoke deployment to remove the hub
- move the log analytics resource to the spoke
- move bastion to the spoke

## HOW to Test?

You can follow the steps from the docs and get this deployed and validated as before.

### CODE REVIEW

File: networking/spoke-BU0001A0008.bicep [hunk 1]
```diff
@@ -30,51 +26,234 @@ param location string
...
-}
+/*** RESOURCES ***/
 
-// This is the firewall that was deployed in 'hub-default.bicep'
-resource hubFirewall 'Microsoft.Network/azureFirewalls@2021-05-01' existing = {
-  scope: hubResourceGroup
-  name: 'fw-${location}'
+// This Log Analytics workspace stores logs from the regional spokes network, and bastion.
+resource la 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
+  name: 'la-${location}'
```
Comment (josevarela) [inline networking/spoke-BU0001A0008.bicep:33]: General comment that we could address on this PR or later: We should do a complete pass of the resource naming conventions to match CAF recommendations. For example, "log" is the recommended prefix for Log Analytics Workspaces

Also, I think the base name should reflect the workload name and not the location (open to discuss)

File: networking/spoke-BU0001A0008.bicep [hunk 2]
```diff
@@ -30,51 +26,234 @@ param location string
...
-  name: '${last(split(hubVnetResourceId,'/'))}'
-}
+/*** RESOURCES ***/
 
-// This is the firewall that was deployed in 'hub-default.bicep'
-resource hubFirewall 'Microsoft.Network/azureFirewalls@2021-05-01' existing = {
-  scope: hubResourceGroup
-  name: 'fw-${location}'
+// This Log Analytics workspace stores logs from the regional spokes network, and bastion.
+resource la 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
```
Comment (josevarela) [inline networking/spoke-BU0001A0008.bicep:32]: General comment that we could address on this PR or later: We should do a complete pass of the bicep naming conventions for resources 

Example:

This: resource la 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {

could be changed to: resource logAnaliticsWorkspace 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {

**Review by josevarela — COMMENTED** [review]
Looks good!. A couple of small comments on naming convention that we can address in this PR or later.

Notice I didn't spend too much time on the content as we are planning to merge into a single file

File: networking/spoke-BU0001A0008.bicep (hunk 1 above)
Comment (ferantivero) [reply networking/spoke-BU0001A0008.bicep:33]: that's is coming as part of the PR#5.

File: networking/spoke-BU0001A0008.bicep (hunk 2 above)
Comment (ferantivero) [reply networking/spoke-BU0001A0008.bicep:32]: I think we will be able to resolve this one via https://github.com/mspnp/iaas-baseline/pull/1#discussion_r1223404094

Comment (ferantivero) [reply networking/spoke-BU0001A0008.bicep:32]: addressed from e3e5205024315457161d2024de6f2a3a33e4d596

File: networking/spoke-BU0001A0008.bicep (hunk 1 above)
Comment (ferantivero) [reply networking/spoke-BU0001A0008.bicep:33]: addressed from e3e5205024315457161d2024de6f2a3a33e4d596

**Pull request merged**